src/PSR7/SpecFinder.php

```diff
@@ -74,37 +108,49 @@ public function findPathSpec(OperationAddress $addr) : PathItem
         return $pathSpec;
     }
 
-    /**
-     * @return Parameter[]
-     *
-     * @throws NoPath
-     */
-    public function findPathSpecs(OperationAddress $addr) : array
+    private function findPathSimple(OperationAddress $addr) : ?PathItem
     {
-        $spec = $this->findOperationSpec($addr);
+        return $this->openApi->paths->getPath($addr->path());
+    }
 
-        // 1. Collect operation-level params
-        $pathSpecs = [];
+    private function findPathUsingFinder(OperationAddress $addr) : ?PathItem
+    {
+        $finder  = new PathFinder($this->openApi, $addr->path(), $addr->method());
+        $results = $finder->search();
+        if (count($results) === 1) {
+            return $this->findPathSimple($results[0]);
+        }
 
-        foreach ($spec->parameters as $p) {
-            if ($p->in !== 'path') {
-                continue;
-            }
+        return null;
```

Comment:
It could turn out that there are multiple matched paths (more than one). returning null here means throwing `NoPath` exception later. Sound non-intuitive. I think we should check this case too